hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/ESP8266WebServer.h

```diff
@@ -83,6 +83,13 @@ class ESP8266WebServer
   String argName(int i);          // get request argument name by number
   int args();                     // get arguments count
   bool hasArg(const char* name);  // check if argument exists
+  
+  void collectHeaders(const char* headerKeys[], const size_t headerKeysCount); // set the request headers to collect
+  String header(const char* name);   // get request header value by name
```

Comment:
Will this end up copying the returned String (or will the compiler optimize)? Is there a "best practice" (examples in other well-used libraries) on how to return a String efficiently?

(Am I focusing _too_ much on optimizations :smile: :disappointed: ?)